doc/application/index.rst

```diff
@@ -273,24 +272,33 @@ should know about.
   :makevar:`ZEPHYR_BASE` explicitly, but then you won't get the other features
   provided by those scripts.
 
-* :makevar:`BOARD`: Selects the board that the application's build will use for
-  the default configuration. This can be defined in the environment, in your
-  application's :file:`CMakeLists.txt` file, or in the ``cmake`` command line.
-  See :ref:`boards` for built-in boards, and :ref:`board_porting_guide` for
-  information on adding board support.
+  If you build your application using :ref:`west build <west-building>`, you
+  don't need to set :makevar:`ZEPHYR_BASE` or source those scripts; west does
+  it for you in the environment ``cmake`` is called in.
```

Comment:
IMO we should always require sourcing and setting ZEPHYR_BASE until we decide to completely remove it. If you do not have ZEPHYR_BASE set lots of other things will not work that expect ZEPHYR_BASE, so it does not make sense to not require it for some command and require it for others, this is just confusing.

Reply:
OK, that's reasonable. As you know, requiring users to set ZEPHYR_BASE themselves causes lots of problems and confusion as we saw at the hackathon, so I think we should definitely eliminate the need for it everywhere we can.

But I will drop that from this patch for now until we can have the bigger conversation.

Reply:
I believe `west` prints a warning and discards the user-defined ZEPHYR_BASE if and only if different from what it automatically inferred, correct?

Reply:
It used to, but it uses the environment variable now. It still warns:

```
$ west build
WARNING: ZEPHYR_BASE=somewhere-else in the calling environment will be used, but was set to /home/mbolivar/zephyr/zephyr in west config.
 To disable this warning, execute 'west config --global zephyr.base-prefer env'
```

You can also set `zephyr.base-prefer` to `configfile` to always get it from the west config's `zephyr.base` instead, which is initialized based on the manifest's `zephyr` repository.